File: client/python/ff_client/ff_request.py
class FfRequest:
    class Version:
        V1 = 1 

    class EncryptionMode:
        AES_256_GCM = 1

    class KeyDeriveMode:
        PBKDF2 = 1

    class Option:
        class Type:
            EOL = 0
            ENCRYPTION_MODE = 1
            ENCRYPTION_IV = 2
            ENCRYPTION_TAG = 3
            HTTPS = 4
            KEY_DERIVE_MODE = 5
            KEY_DERIVE_SALT = 6
            BREAK = 7
            TIMESTAMP = 8
        
        def __init__(self, type: int, length: int, value: bytearray):
            self.type = type
            self.length = length
            self.value = value

    def __init__(self, version: int = None, request_id: int = None):
        self.version = version
        self.request_id = request_id
        self.options = []
        self.secure_options = []
        self.payload = None

File: client/java/src/main/java/com/timetoogo/ffclient/FfRequest.java

package com.timetoogo.ffclient;

import java.util.ArrayList;
import java.util.List;

public class FfRequest {
    public enum Version {
        V1((short) 1);

        private short value;

        private Version(short value) {
            this.value = value;
        }

        public short getValue() {
            return value;
        }
    }

    public enum EncryptionMode {
        AES_256_GCM((byte) 1);

        private byte value;

        private EncryptionMode(byte value) {
            this.value = value;
        }

        public byte getValue() {
            return value;
        }
    }

    public enum KeyDeriveMode {
        PBKDF2((byte) 1);

        private byte value;

        private KeyDeriveMode(byte value) {
            this.value = value;
        }

        public byte getValue() {
            return value;
        }
    }

    private Version version;
    private long requestId;
    private List<FfRequestOption> options = new ArrayList<FfRequestOption>();
    private byte[] payload;

    public Version getVersion() {
        return version;
    }

    public void setVersion(Version version) {
        this.version = version;
    }

    public long getRequestId() {
        return requestId;
    }

    public void setRequestId(long requestId) {
        this.requestId = requestId;
    }

    public List<FfRequestOption> getOptions() {
        return options;
    }

    public void setOptions(List<FfRequestOption> options) {
        this.options = options;
    }

    public byte[] getPayload() {
        return payload;
    }

    public void setPayload(byte[] payload) {
        this.payload = payload;
    }
}
